src/cmds/testing/memtest/memtest.c

```diff
@@ -12,27 +12,143 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdint.h>
+#include <stdbool.h>
 #include <unistd.h>
 #include <time.h>
 
 #include <mem/vmem.h>
 
-static const size_t default_region_len = 16 * 1024 * 1024;
-
 static void print_help(char **argv) {
 	assert(argv);
 	assert(argv[0]);
 	printf("Usage: %s [-nh] [-l LENGHT]\n", argv[0]);
 	printf("\tOptions: -n Test non-cached memory\n");
-	printf("\t         -l LENGTH Number of bytes to be copied (default 0x1000000)\n");
-	printf("\t         -r REPEAT Number of repeats (default 64)\n");
+	printf("\t         -l LENGTH Number of bytes to be operated "
+			"(default 0x1000000 for memcpy() and 0x2000 for heap)\n");
+	printf("\t         -r REPEAT Number of repeats (default 64 for memcpy() and 2048 for heap)\n");
+	printf("\t         -p Run test for heap (a.k.a. pyramid)\n");
+	printf("\t         -q Maximum number of heap regions\n");
+	printf("\t         -s Random seed\n");
+}
+
+/* Heap-related stuff */
+#define MAX_HEAP_REGIONS 4096
+
+static int max_region_len;
```

Comment:
may be create a structure with global fields and pass it in routines as argument?